main.js

```diff
@@ -127,5 +138,115 @@ for (var i = 0 ; i < courses.length ; i++) {
 
 
 
+//add button completed with checkbox that will show completed
+//uncompleted courses based on its status
+
+
+AllcoursesBtn.onclick =function (){
+   display (courses); 
+}
+
+function completed(i){
+
+    let checkBox = document.getElementById(`course${i}`)
+    if (checkBox && checkBox.checked ){
+
+         courses[i].IsCompleted=true;
+         localStorage.setItem('courses',JSON.stringify(courses))
+
+    }
+    else{
+        courses[i].IsCompleted=false;
+        localStorage.setItem('courses',JSON.stringify(courses))
+
+    }
+}
+
+function completed(i, prefix) {
+    let checkBox = document.getElementById(`${prefix}${i}`);
+    
+    if (checkBox && checkBox.checked) {
+        courses[i].IsCompleted = true;
+        localStorage.setItem('courses', JSON.stringify(courses));
+    } else {
+        courses[i].IsCompleted = false;
+        localStorage.setItem('courses', JSON.stringify(courses));
+    }
+}
```

Comment:
this is the same as the above function, can you reuse it to support both cases with prefix and without